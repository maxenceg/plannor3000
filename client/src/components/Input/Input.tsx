// @flow
import React, { ChangeEvent } from 'react';
import { TextField } from '@material-ui/core';

interface Props {
  label?: string | null;
  error?: string | null;
  type: string;
  disabled?: boolean;
  field: {
    name?: string;
    onBlur?: () => any;
    onChange: (
      event: ChangeEvent<HTMLInputElement | HTMLSelectElement | HTMLTextAreaElement>,
    ) => void;
    value?: string;
  };
}

const Input: React.SFC<Props> = props => {
  const { error, field, ...otherProps } = props;

  return (
    <TextField
      variant="outlined"
      margin="normal"
      error={!!error}
      helperText={error || null}
      {...field}
      {...otherProps}
    />
  );
};

export default Input;
