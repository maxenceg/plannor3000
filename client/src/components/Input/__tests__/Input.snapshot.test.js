// @flow
import React from 'react';
import { shallow } from 'enzyme';

import Input from '..';

describe('<Input>', () => {
  it('should render correctly', () => {
    const props = {
      name: 'Passphrase',
      label: 'French atomic bomb passphrase',
      type: 'password',
      field: {
        value: 'La barbe de la femme à Georges Moustaki',
        onChange: value => null,
      }
    };
    const tree = shallow(<Input {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with an error provided', () => {
    const props = {
      name: 'Passphrase',
      label: 'French atomic bomb passphrase',
      type: 'password',
      error: 'Password seems too long to be true',
      field: {
        value: 'La barbe de la femme à Georges Moustaki',
        onChange: value => null,
      }
    };
    const tree = shallow(<Input {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
