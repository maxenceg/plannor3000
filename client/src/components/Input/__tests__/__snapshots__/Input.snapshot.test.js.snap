// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input> should render correctly 1`] = `
<TextField
  error={false}
  helperText={null}
  label="French atomic bomb passphrase"
  margin="normal"
  name="Passphrase"
  onChange={[Function]}
  required={false}
  select={false}
  type="password"
  value="La barbe de la femme à Georges Moustaki"
  variant="outlined"
/>
`;

exports[`<Input> should render correctly with an error provided 1`] = `
<TextField
  error={true}
  helperText="Password seems too long to be true"
  label="French atomic bomb passphrase"
  margin="normal"
  name="Passphrase"
  onChange={[Function]}
  required={false}
  select={false}
  type="password"
  value="La barbe de la femme à Georges Moustaki"
  variant="outlined"
/>
`;
